modules/server/src/main/java/org/jboss/wsf/stack/cxf/deployment/WSDLFilePublisher.java

```diff
@@ -118,28 +118,53 @@ public void publishWsdlFiles(QName serviceName, String wsdlLocation, Bus bus, Li
 
    public void unpublishWsdlFiles()
    {
-      try
+      String deploymentName = dep.getCanonicalName();
+
+      if (deploymentName.startsWith("http://"))
       {
-         File parentDir = new File(serverConfig.getServerDataDir().getCanonicalPath() + "/wsdl");
-         ArchiveDeployment deployment = dep;
-         while (deployment.getParent() != null)
-         {
-            deployment = deployment.getParent();
-         }
-         String deploymentName = deployment.getCanonicalName();
-         if (deploymentName.startsWith("http://"))
-         {
-            deploymentName = deploymentName.replace("http://", "http-");
-         }
-         File targetDir = new File(parentDir, deploymentName);
-         FileUtils.removeDir(targetDir);
+         deploymentName = deploymentName.replace("http://", "http-");
       }
-      catch (IOException e)
-      {
+
+      try {
+
+         File wsdlRootDir = new File(serverConfig.getServerDataDir().getCanonicalPath()
+             + "/wsdl/");
+         File ancestorDir = new File(serverConfig.getServerDataDir().getCanonicalPath()
+             + "/wsdl/" + deploymentName);
+
+         // create directory list to be deleted
+         List<File> dirList = new ArrayList<File>();
+         String wsdlRootDirStr = wsdlRootDir.getCanonicalPath();
+         do {
+            dirList.add(ancestorDir);
+            ancestorDir = ancestorDir.getParentFile();
+         } while(!wsdlRootDirStr.equals(ancestorDir.getAbsolutePath()));
+
+         for (File publishDir : dirList) {
+            if (publishDir.exists()) {
+               File[] wsdlFileList = publishDir.listFiles();
+               if (wsdlFileList != null) {
+                  for (int i = 0; i < wsdlFileList.length; i++) {
+                     File f = wsdlFileList[i];
+                     if (f.delete()) {
```

Comment:
what if the file is a directory and is not empty? IOW, you're assuming here that each directory is nested within a single directory only.